tests/Middleware/ActionCallerTest.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Yiisoft\Yii\Web\Tests\Middleware;
+
+use Nyholm\Psr7\Response;
+use Psr\Container\ContainerInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Yiisoft\Yii\Web\Middleware\ActionCaller;
+use PHPUnit\Framework\TestCase;
+
+class ActionCallerTest extends TestCase
+{
+    /** @var ServerRequestInterface  */
+    private $request;
+
+    /** @var RequestHandlerInterface  */
+    private $handler;
+
+    protected function setUp()
+    {
+        $this->request = $this->createMock(ServerRequestInterface::class);
+        $this->handler = $this->createMock(RequestHandlerInterface::class);
+    }
+
+    public function testProcess()
+    {
+        $container = $this->createMock(ContainerInterface::class);
+        $container->method('get')->willReturn($this);
+
+        $caller = new ActionCaller('this', 'process', $container);
+
+        $response = $caller->process($this->request, $this->handler);
+        self::assertEquals(204, $response->getStatusCode());
+    }
+
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
```

Comment:
Now it verifies that request and handler are correctly passed 👍

Let's expand it to type-hinting for a custom service to verify that this service will be searched in DI container.

Reply:
Is it acceptable changes?
![image](https://user-images.githubusercontent.com/289431/67957236-5c4c0500-fc06-11e9-9b4c-a441a7f7762c.png)

Reply:
Not mocking container? Yes. Absolutely. That's actually way cleaner than using a mock.